src/Loader.ts

```diff
@@ -0,0 +1,132 @@
+export const RACETIME = "https://racetime.gg";
+
+export type UserData = {
+    id: string;
+    full_name: string;
+    name: string;
+    discriminator: string;
+    url: string;
+    avatar: string;
+    pronouns: string;
+    flair: string;
+    twitch_name: string;
+    twitch_display_name: string;
+    twitch_channel: string;
+    can_moderate: boolean;
+};
+
+export enum EntrantStatus {
+    Done = "done",
+    DNF = "dnf",
+}
+
+export type RaceData = {
+    name: string;
+    status: {
+        value: string;
+        // verbose: string,
+        // help_text: string,
+    };
+    goal: {
+        name: string;
+        custom: boolean;
+    };
+    info: string;
+    entrants_count: number;
+    // entrants_count_finished: number,
+    // entrants_count_inactive: number,
+    // opened_at: string,
+    ended_at: string;
+    recorded: boolean;
+    entrants: Array<{
+        user: UserData;
+        status: {
+            value: EntrantStatus;
+        };
+        finish_time: string;
+        place: number;
+        comment: string;
+    }>;
+};
+
+type RaceListData = {
+    count: number;
+    num_pages: number;
+    races: Array<RaceData>;
+};
+
+export type SlugData = {
+    name: string;
+    short_name: string;
+    slug: string;
+    url: string;
+    data_url: string;
+    image: string;
+    info: string;
+    streaming_required: boolean;
+    owners: Array<UserData>;
+    moderators: Array<UserData>;
+    goals: Array<string>;
+    current_races: Array<RaceData>;
+    emotes: Map<string, string>;
```

Comment:
A browser's JSON decoder will not create a Map object, this is probably meant to be a `Record<string, string>`.